src/hints/hint_processor.rs

```diff
@@ -0,0 +1,124 @@
+            BOOTLOADER_GUESS_PRE_IMAGE_OF_SUBTASKS_OUTPUT_HASH => {
+                guess_pre_image_of_subtasks_output_hash(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            BOOTLOADER_PREPARE_SIMPLE_BOOTLOADER_OUTPUT_SEGMENT => {
+                prepare_simple_bootloader_output_segment(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            BOOTLOADER_COMPUTE_FACT_TOPOLOGIES => {
+                compute_and_configure_fact_topologies(vm, exec_scopes)
+            }
+            BOOTLOADER_SET_PACKED_OUTPUT_TO_SUBTASKS => set_packed_output_to_subtasks(exec_scopes),
+            BOOTLOADER_IMPORT_PACKED_OUTPUT_SCHEMAS => import_packed_output_schemas(),
+            BOOTLOADER_IS_PLAIN_PACKED_OUTPUT => is_plain_packed_output(vm, exec_scopes),
+            BOOTLOADER_ASSERT_IS_COMPOSITE_PACKED_OUTPUT => {
+                assert_is_composite_packed_output(exec_scopes)
+            }
+            SIMPLE_BOOTLOADER_PREPARE_TASK_RANGE_CHECKS => {
+                prepare_task_range_checks(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            SIMPLE_BOOTLOADER_SET_TASKS_VARIABLE => set_tasks_variable(exec_scopes),
+            SIMPLE_BOOTLOADER_DIVIDE_NUM_BY_2 => divide_num_by_2(vm, ids_data, ap_tracking),
+            SIMPLE_BOOTLOADER_SET_CURRENT_TASK => {
+                set_current_task(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            SIMPLE_BOOTLOADER_ZERO => set_ap_to_zero(vm),
+            EXECUTE_TASK_ALLOCATE_PROGRAM_DATA_SEGMENT => {
+                allocate_program_data_segment(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            EXECUTE_TASK_LOAD_PROGRAM => load_program_hint(vm, exec_scopes, ids_data, ap_tracking),
+            EXECUTE_TASK_VALIDATE_HASH => validate_hash(vm, exec_scopes, ids_data, ap_tracking),
+            EXECUTE_TASK_ASSERT_PROGRAM_ADDRESS => {
+                assert_program_address(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            EXECUTE_TASK_CALL_TASK => call_task(vm, exec_scopes, ids_data, ap_tracking),
+            EXECUTE_TASK_WRITE_RETURN_BUILTINS => {
+                write_return_builtins_hint(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            EXECUTE_TASK_EXIT_SCOPE => exit_scope(exec_scopes),
+            EXECUTE_TASK_APPEND_FACT_TOPOLOGIES => {
+                append_fact_topologies(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            SELECT_BUILTINS_ENTER_SCOPE => {
+                select_builtins_enter_scope(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            INNER_SELECT_BUILTINS_SELECT_BUILTIN => {
+                select_builtin(vm, exec_scopes, ids_data, ap_tracking)
+            }
+            unknown_hint_code => Err(HintError::UnknownHint(
+                unknown_hint_code.to_string().into_boxed_str(),
+            )),
+        }
+    }
+}
+
+impl ResourceTracker for BootloaderHintProcessor {}
```

Comment:
Something is weird, this is duplicated from `main`. I think there was some kind of issue during your rebase. I think you can remove this.